x-pack/plugins/security_solution/public/timelines/components/edit_data_provider/index.tsx

```diff
@@ -277,3 +303,127 @@ export const StatefulEditDataProvider = React.memo<Props>(
 );
 
 StatefulEditDataProvider.displayName = 'StatefulEditDataProvider';
+
+function isStringOrNumberArray(
+  val: string | number | Array<string | number>
+): val is Array<string | number> {
+  return Array.isArray(val) && (typeof val[0] === 'string' || typeof val[0] === 'number');
+}
+
+const isValueFieldInvalid = (type: DataProviderType, value: string | number) =>
```

Comment:
consider extracting this function out of the component and adding unit tests for it

Reply:
done!